### ISSUE TITLE
Feature idea: Having connected plugin name instead of AudioGridder/AGridder visible in DAW

### ISSUE THREAD

**Issue by igbro** [apohl79/audiogridder#253]
I know this is not part of VST protocol to allow changing the apparent plugin name in the DAW. 
Don't know for the others formats, sorry.

For VST's, the tweak is, like Acustica Nebula is allowing, to create copies of AG plugin DLL/VST3 files.

The problem is that simply renaming the file wont make it, because most DAWs use VST unique Id to identify plugins not the file name.

This can be done manually with an utility to reveal Id and an Hex editor, but is a bit tricky (like trying not to collide with an Id you already have in your VST pool, and Id is also often coded in reverse mode)

See this old SOS article about ID "Hacking"
https://www.soundonsound.com/sound-advice/resolving-vst-plug-conflicts

So for this purpose, we will need an utility (like N4 offers) to create a copy with the chosen name and an unique ID.

The beauty would be if this new named instance of AG plugin could remember the connected plugins and be abbe to reconnect to those at launch.

Just saying...

**Comment by jaacMusic** [comment]
How do you deal with VST chain display? I see your reasoning as it would help identify which plugins are loaded before having to open AG plugin instance to see if x VST is loaded on x channel.

**Comment by apohl79** [comment]
Interesting idea. Instead of just saving the chain as a named preset (I will look at this in 1.2) you would basically create a chain as new named plugin. I will think about it. Thx.

**Comment by igbro** [comment]
JaacMusic, naming is free. You could name your chain as you wish.

Sorry if I'm not super clear, English is not my first language.

I already made this manually with two different versions of Kontakt 5 which normally can't cohabit in the DAW because they are sharing the same ID. 

Yes, it would let you identify what's in there (you choose the name) without opening the AG plugin interface and be able to directly choose in your VST pool a plugin with a name meaning something for you.

Imagine you want to have a deported Nebula 4 directly accessible in your VST pool.
Let's create a copy in your VST directory of AG plugin named for instance "N4 AG" and give this a unique VST ID.
After a plugin scan, your DAW will let you access this new plugin and the shown name will be N4 AG.

For now, this can already be done manually, but it will load blank like a regular Ag instance. You will still have to open the AG UI to choose and connect to your plugin(s), having the only small benefit to have a recognizable name (if you effectively connect to an deported N4)

OK, meanwhile Andreas as commented! He got it, ahahah!

Yes, to be efficient, this new Ag instance will have to recall it's "preset" and reconnect automatically to the saved plugin chain at launch.

**Comment by igbro** [comment]
Andreas, you got it!

**Comment by igbro** [comment]
This is how it works in Nebula 4

![N4 setups](https://user-images.githubusercontent.com/65025818/101288379-87842900-37f6-11eb-80c1-a3e85a9e946e.JPG)

**Comment by CaBuffalo** [comment]
Maybe AG plugin name as default. User can name it as wish. 👍

**Comment by igbro** [comment]
Yes, but you can't have several ones with the same ID in your VST folders, almost all DAWs will see only one of them at rescan.

By the way, one downside off all this is the necessity for a rescan. 
Which means, you won't have your fresh new plugin (saved chain preset) immediately available after creation.

**Comment by CaBuffalo** [comment]
What about showing a message of the names of plugins in the chain once you put mouse hover it ?

**Comment by igbro** [comment]
Would be great but it's not possible.
A plugin is hosted and it's code is restricted to only operate at it's level.
Name showing, and how you can interact with it is part of host code.

The trick to have several copies of the AG plugin with different names/IDs is exactly because you can't through the VST protocol, freely name or rename dynamically what is shown in the DAW.

Can't speak for AU.
Vst4?
